packages/core/src/internal/proc.js

```diff
@@ -473,7 +473,14 @@ export default function proc(
     } else if (is.func(promise.abort)) {
       cb.cancel = () => promise.abort()
     }
-    promise.then(cb, error => cb(error, true))
+    promise.then(cb, error => {
+      if (error instanceof Error) {
```

Comment:
I dont think we should actually convert those not-errors to errors. Let them flow through the system and guard against them at later point - before attaching properties to them and before logging them